apps/managedidentity/managedidentity_test.go

```diff
@@ -48,95 +79,130 @@ func getSuccessfulResponse(resource string) ([]byte, error) {
+		t.Setenv(IdentityEndpointEnvVar, "identityEndpointEnvVar value")
+		t.Setenv(IdentityHeaderEnvVar, "identityHeaderEnvVar value")
+	case CloudShell:
+		t.Setenv(MsiEndpointEnvVar, "msiEndpointEnvVar value")
+	case ServiceFabric:
+		t.Setenv(IdentityEndpointEnvVar, "identityEndpointEnvVar value")
+		t.Setenv(IdentityHeaderEnvVar, "identityHeaderEnvVar value")
+		t.Setenv(IdentityServerThumbprintEnvVar, "identityServerThumbprintEnvVar value")
+	}
+}
+
+func unsetEnvVars() {
+	os.Unsetenv(IdentityEndpointEnvVar)
+	os.Unsetenv(IdentityHeaderEnvVar)
+	os.Unsetenv(IdentityServerThumbprintEnvVar)
+	os.Unsetenv(ArcIMDSEnvVar)
+	os.Unsetenv(MsiEndpointEnvVar)
+}
+
+func Test_Get_Source(t *testing.T) {
+	// todo update as required
+	testCases := []sourceTestData{
+		{name: "testAzureArcSystemAssigned", source: AzureArc, endpoint: imdsEndpoint, expectedSource: AzureArc, miType: SystemAssigned()},
+		{name: "testAzureArcUserClientAssigned", source: AzureArc, endpoint: imdsEndpoint, expectedSource: AzureArc, miType: UserAssignedClientID("clientId")},
+		{name: "testAzureArcUserResourceAssigned", source: AzureArc, endpoint: imdsEndpoint, expectedSource: AzureArc, miType: UserAssignedResourceID("resourceId")},
+		{name: "testAzureArcUserObjectAssigned", source: AzureArc, endpoint: imdsEndpoint, expectedSource: AzureArc, miType: UserAssignedObjectID("objectId")},
+		{name: "testDefaultToImds", source: DefaultToIMDS, endpoint: imdsEndpoint, expectedSource: DefaultToIMDS, miType: SystemAssigned()},
+		{name: "testDefaultToImdsClientAssigned", source: DefaultToIMDS, endpoint: imdsEndpoint, expectedSource: DefaultToIMDS, miType: UserAssignedClientID("clientId")},
+		{name: "testDefaultToImdsResourceAssigned", source: DefaultToIMDS, endpoint: imdsEndpoint, expectedSource: DefaultToIMDS, miType: UserAssignedResourceID("resourceId")},
+		{name: "testDefaultToImdsObjectAssigned", source: DefaultToIMDS, endpoint: imdsEndpoint, expectedSource: DefaultToIMDS, miType: UserAssignedObjectID("objectId")},
+		{name: "testDefaultToImdsEmptyEndpoint", source: DefaultToIMDS, endpoint: "", expectedSource: DefaultToIMDS, miType: SystemAssigned()},
+		{name: "testDefaultToImdsLinux", source: DefaultToIMDS, endpoint: imdsEndpoint, expectedSource: DefaultToIMDS, miType: SystemAssigned()},
+		{name: "testDefaultToImdsEmptyEndpointLinux", source: DefaultToIMDS, endpoint: "", expectedSource: DefaultToIMDS, miType: SystemAssigned()},
 	}
 
 	for _, testCase := range testCases {
-		t.Run(http.StatusText(testCase.code), func(t *testing.T) {
-			fakeErrorClient := mock.Client{}
-			responseBody, err := makeResponseWithErrorData(testCase.err, testCase.desc)
-			if err != nil {
-				t.Fatalf("error while forming json response : %s", err.Error())
+		t.Run(string(testCase.source), func(t *testing.T) {
+			unsetEnvVars()
+			setEnvVars(t, testCase.source)
+
+			if runtime.GOOS == "linux" {
+				originalPath := azureArcOsToFileMap[runtime.GOOS]
+				azureArcOsToFileMap[runtime.GOOS] = "fake/fake"
```

Comment:
Generally best not to modify global state in a test and when you must, to restore the prior value after the test